Cython/Compiler/Main.py

```diff
@@ -623,9 +623,24 @@ def compile_multiple(sources, options):
     cwd = os.getcwd()
     for source in sources:
         if source not in processed:
+            output_filename = get_output_filename(source, cwd, options)
             if context is None:
                 context = Context.from_options(options)
-            output_filename = get_output_filename(source, cwd, options)
+            if options.cache:
```

Comment:
Shouldn't the caching happen *after* the out of date check?

Reply:
fixed.